sklearn/cluster/_kmeans.py

```diff
@@ -926,6 +946,32 @@ class KMeans(
            [ 1.,  2.]])
     """
 
+    _expected_params_type_and_vals = {
+        "n_clusters": [(numbers.Integral, Interval(1, None, closed="left"))],
+        "init": [
+            (str, {"k-means++", "random"}),
+            (callable,),
```

Comment:
Future work: we can imagine defining the subset of callables with a specific signature to pass here as valid values